Docs/source/install/hpc/pitzer.rst

```diff
@@ -0,0 +1,218 @@
+
+Introduction
+------------
+
+If you are new to this system, **please see the following resources**:
+
+* `Pitzer user guide <https://www.osc.edu/resources/getting_started/new_user_resource_guide>`__
+* Batch system: `Slurm manual <https://www.osc.edu/resources/technical_support/slurm>`__ and `OSC specific guide <https://www.osc.edu/supercomputing/batch-processing-at-osc/job-submission>`__
+* `Jupyter service <https://www.osc.edu/vocabulary/documentation/jupyter>`__
+* `Filesystems <https://www.osc.edu/supercomputing/storage-environment-at-osc/storage-hardware/overview_of_file_systems>`__:
+
+  * ``$HOME``: per-user directory, use only for inputs, source, and scripts; backed up (500GB)
+  * ``/fs/ess``: per-project storage directory, use for long-term storage of data and analysis; backed up (1-5TB)
+  * ``/fs/scratch``: per-project production directory; fast I/O for parallel jobs; not backed up (100TB)
+
+.. _building-pitzer-preparation:
+
+Preparation
+-----------
+
+Use the following commands to download the WarpX source code:
+
+.. code-block:: bash
+
+   git clone https://github.com/ECP-WarpX/WarpX.git $HOME/src/warpx
+
+On Pitzer, you can run either on GPU nodes with V100 GPUs or CPU nodes. They can use the same profile.
+
+We use system software modules, add environment hints, and further dependencies via the file ``$HOME/pitzer_warpx.profile``. Create it now:
+
+.. code-block:: bash
+
+   cp $HOME/src/warpx/Tools/machines/pitzer-osc/pitzer_warpx.profile.example $HOME/pitzer_warpx.profile
+
+.. dropdown:: Script Details
+   :color: light
+   :icon: info
+   :animate: fade-in-slide-down
+
+   .. literalinclude:: ../../../../Tools/machines/pitzer-osc/pitzer_warpx.profile.example
+      :language: bash
+
+Edit the 2nd line of this script, which sets the ``export proj=""`` variable. For example, if you are a member of the project ``pas2024``, then run ``nano $HOME/pitzer_warpx.profile`` and edit line 2 to read:
+
+.. code-block:: bash
+
+   export proj="pas2024"
+
+Exit the ``nano`` editor with ``Ctrl`` + ``O`` (save) and then ``Ctrl`` + ``X`` (exit).
+
+.. important::
+
+   Now, and as the first step on future logins to Pitzer, activate these environment settings:
+
+   .. code-block:: bash
+
+      source $HOME/pitzer_warpx.profile
+
+Finally, since Pitzer does not yet provide software modules for some of our dependencies, install them once:
+
+.. code-block:: bash
+
+   bash $HOME/src/warpx/Tools/machines/pitzer-osc/install_v100_dependencies.sh
+   source $HOME/sw/pitzer/v100/venvs/warpx-pitzer/bin/activate
+
+.. dropdown:: Script Details
+   :color: light
+   :icon: info
+   :animate: fade-in-slide-down
+
+   .. literalinclude:: ../../../../Tools/machines/pitzer-osc/install_dependencies.sh
+      :language: bash
+
+.. _building-pitzer-compilation:
+
+Compilation
+-----------
+
+Use the following :ref:`cmake commands <building-cmake>` to compile the application executable:
+
+.. tab-set::
+   .. tab-item:: V100 GPUs
```

Comment:
The compilation shows a V100 GPU and a CPU version, but the profile and dependency install is only for GPU. This way, building for CPU will not work well, because the dependencies like BLAS++ are build for GPU (and not for CPU).

If you want to document both CPU and GPU usage, I would recommend to have separate profiles for GPU and CPU and separate install dependency scripts for GPU and CPU. See Perlmutter (NERSC) for an example.

Reply:
Actually they can work together but that is indeed a good practice. I have separated the GPU and CPU profiles and scripts for installing dependencies.